src/de/willuhn/jameica/fibu/gui/action/BuchungSplitNeu.java

```diff
@@ -0,0 +1,87 @@
+package de.willuhn.jameica.fibu.gui.action;
+
+import java.rmi.RemoteException;
+
+import de.willuhn.datasource.rmi.DBIterator;
+import de.willuhn.jameica.fibu.Fibu;
+import de.willuhn.jameica.fibu.Settings;
+import de.willuhn.jameica.fibu.rmi.Buchung;
+import de.willuhn.jameica.fibu.rmi.HilfsBuchung;
+import de.willuhn.jameica.gui.Action;
+import de.willuhn.jameica.gui.GUI;
+import de.willuhn.jameica.system.Application;
+import de.willuhn.logging.Logger;
+import de.willuhn.util.ApplicationException;
+import de.willuhn.util.I18N;
+
+/**
+ * Action zum Splitten einer Buchung.
+ * @author henken
+ */
+public class BuchungSplitNeu implements Action
+{
+
+  /**
+   * @see de.willuhn.jameica.gui.Action#handleAction(java.lang.Object)
+   */
+  public void handleAction(Object context) throws ApplicationException
+  {
+	Buchung buchung = null;
+	Buchung hauptBuchung = null;
+    if (context != null)
+    {
+    	try
+        {
+	      if (context instanceof Buchung) {
+	    	  if(((Buchung)context).getSplitHauptBuchung() != null)
+	    		  hauptBuchung = ((Buchung)context).getSplitHauptBuchung();
+	    	  else
+	    		  hauptBuchung = (Buchung) context;
+	      }
+	      else if (context instanceof HilfsBuchung)
+	    	  hauptBuchung = ((HilfsBuchung)context).getHauptBuchung();
+        }
+        catch (RemoteException e)
+        {
+          Logger.error("unable to load hauptbuchung",e);
+          I18N i18n = Application.getPluginLoader().getPlugin(Fibu.class).getResources().getI18N();
+          throw new ApplicationException(i18n.tr("Fehler beim Laden der Buchung"));
+        }
+		try {
+			//Einige Werte der Hauptbuchung �bernehmen
+			buchung = (Buchung) Settings.getDBService().createObject(Buchung.class,null);
+	        buchung.setGeschaeftsjahr(Settings.getActiveGeschaeftsjahr());
+	        
+	        buchung.setBelegnummer(hauptBuchung.getBelegnummer());
+	        buchung.setDatum(hauptBuchung.getDatum());
+	        buchung.setHibiscusUmsatzID(hauptBuchung.getHibiscusUmsatzID());
+	        buchung.setText(hauptBuchung.getText());
+	        buchung.setHabenKonto(hauptBuchung.getHabenKonto());
+	        buchung.setSollKonto(hauptBuchung.getSollKonto());
+	        buchung.setSteuer(hauptBuchung.getSteuer());
+	        buchung.setGeprueft(hauptBuchung.isGeprueft());
+	        
+	        //Betrag aller bisherigen SplitBuchungen berechnen, daraus den Restbetrag errechnen
+	        double betrag = 0d;
+	        DBIterator i = hauptBuchung.getSplitBuchungen();
+	        while(i.hasNext())
+	        {
+	        	Buchung b = (Buchung) i.next();
+		       betrag +=  b.getBruttoBetrag();
+	        }
+	        double betrag_alt = Double.parseDouble("0"+hauptBuchung.getKommentar());
```

Comment:
Das hier verstehe ich nicht. Warum der Kommentar? Selbst wenn da nichts drin steht, erzeugt das ein "0null" und damit eine NumberFormatException.

Reply:
Ich habe das Kommentar Feld missbraucht um den ursprünglichen Betrag der Buchung zu speichern. Korrekter wäre natürlich das Betrag Feld zu verwenden, das ist jedoch komplizierter, da ich ja dann den Brutto-Betrag ohne Splitbuchungen holen muss. 
Um die Exception zu umgehen hatte ich in AbstractBaseBuchungImpl getKommentar() geändert, hatte ich beim Commit vergessen.
```
public String getKommentar() throws RemoteException
  {
	  Object s = this.getAttribute("kommentar");
	  if(s == null)
		  return "";
	  return (String) s;
  }
```

Reply:
Ich habe es nun so geändert, dass das Betrag Feld verwendet wird